fuse/dir.go

```diff
@@ -44,31 +44,31 @@ func (d *dir) Lookup(ctx context.Context, req *fuse.LookupRequest, resp *fuse.Lo
 		return nil, fuse.ENOENT
 	}
 
-	for _, entry := range entries {
-		cname := plugin.CName(entry)
-		if cname == req.Name {
-			log.Infof("FUSE: Find[d,pid=%v] %v/%v", req.Pid, d, cname)
-			if plugin.ListAction.IsSupportedOn(entry) {
-				childdir := newDir(entry.(plugin.Group))
-				journal.Record(ctx, "FUSE: Found directory %v", childdir)
-				// Prefetch directory entries into the cache
-				go func() {
-					// Need to use a different context here because we still want the prefetch
-					// to happen even when the current context is cancelled.
-					jid := journal.PIDToID(int(req.Pid))
-					ctx := context.WithValue(context.Background(), journal.Key, jid)
-					_, err := childdir.children(ctx)
-					journal.Record(ctx, "FUSE: Prefetching children of %v complete: %v", childdir, err)
-				}()
-				return childdir, nil
-			}
-
-			journal.Record(ctx, "FUSE: Found file %v/%v", d, cname)
-			return newFile(entry), nil
-		}
+	cname := req.Name
+	entry, ok := entries[cname]
+	if !ok {
+		journal.Record(ctx, "FUSE: %v not found in %v", req.Name, d)
+		return nil, fuse.ENOENT
 	}
-	journal.Record(ctx, "FUSE: %v not found in %v", req.Name, d)
-	return nil, fuse.ENOENT
+
+	log.Infof("FUSE: Find[d,pid=%v] %v/%v", req.Pid, d, cname)
+	if plugin.ListAction.IsSupportedOn(entry) {
+		childdir := newDir(d.entry.(plugin.Group), entry.(plugin.Group))
+		journal.Record(ctx, "FUSE: Found directory %v", childdir)
+		// Prefetch directory entries into the cache
```

Comment:
Do we still need to prefetch things? Would be nice to remove this logic if the answer's no so that the logs don't get cluttered up.

Reply:
We could try removing it. I think it potentially makes the interface more responsive, but may not actually be a useful trade-off. Worth testing out.

Reply:
Ok, can do that later. It's not too bad right now